devito/finite_differences/finite_difference.py

```diff
@@ -366,3 +372,51 @@ def generate_fd_shortcuts(function):
             derivatives[name_fd] = (deriv, desciption)
 
     return derivatives
+
+
+def symbolic_weights(function, deriv_order, indices, dim):
+
+    n_weights = len(indices)
+
+    weights = []
+    for j in range(n_weights):
+        weights.append(function.fd_coeff_symbol(indices[j], deriv_order, function, dim))
+    return weights
+
+
+def generate_indices(func, dim, diff, order, stagger=None, side=None):
+
+    # Check if called from first_derivative()
+    if bool(side):
+        if side == right:
+            ind = [(dim+i*diff) for i in range(-int(order/2)+1-(order % 2),
+                                               int((order+1)/2)+2-(order % 2))]
+        elif side == left:
+            ind = [(dim-i*diff) for i in range(-int(order/2)+1-(order % 2),
+                                               int((order+1)/2)+2-(order % 2))]
+        else:
+            ind = [(dim+i*diff) for i in range(-int(order/2),
+                                               int((order+1)/2)+1)]
+        x0 = None
+    else:
+        if stagger == left or not func.is_Staggered:
```

Comment:
should we move these lines down within ``if func.is_Staggered`` ?

Reply:
Yeah, I see no reason not too. And we can also do away with the `or not func.is_staggered` part.